onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -2257,7 +2257,7 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<FusedNodeAnd
           engine_file.read((char*)engine_buf.get(), engine_size);
           *(trt_state->engine) = std::unique_ptr<nvinfer1::ICudaEngine>(
               trt_state->runtime->deserializeCudaEngine(engine_buf.get(), engine_size, nullptr));
-          if (trt_state->engine == nullptr) {
+          if (*(trt_state->engine) == nullptr) {
```

Comment:
The engine is a pointer to unique_ptr. 
https://github.com/microsoft/onnxruntime/blob/a8c313dec41150e8930a6154ad9ced687958f543/onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.h#L104

This is a strange design. Shall it be changed to unique_ptr instead?